src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -186,10 +176,18 @@ public AppPage navigateTo(Url url) {
      * the type indicated by the parameter {@code newPageType}.
      */
     public <T extends AppPage> T changePageType(Class<T> newPageType) {
-        return createNewPage(browser, newPageType);
+        return getNewPageInstance(browser, newPageType);
     }
 
     /**
+     * Gives a LoginPage instance based on the given Browser and test configuration.
```

Comment:
@JosephineKwa I'll add "Fails if the new page is not a valid login page" here later.